docs/docs/propeller/index.md

```diff
@@ -0,0 +1,40 @@
+---
+sidebar_position: 4
+---
+
+# Propeller
+
+Propeller is EquinoxJS's "[propulsion](https://github.com/jet/propulsion)
+lite". It's a set of abstractions for building reactive event processing
+pipelines catering specifically for event sourced reactions from Equinox's
+stores.
+
+## StreamsSink
+
+The `StreamsSink` receives batches of events from a source and intelligently
+manages handling them.
+
+- **Stream management**: maintains an ordered queue of incoming streams,
+  ensuring that reactions processing aligns with the ordering of the triggering
+  events; the aim being to be able to checkpoint attained progress ('tick off
+  batches') as soon as possible.
+- **Concurrent stream handling**: To ensure optimal performance, `StreamsSink`
+  restricts the number of streams that are processing at any given moment to a
+  configurable limit. Additionally guarantees that each stream can have at most
```

Comment:
This is not a lil bonus - its a central thing

>Additionally guarantees that each stream can have at most
  one handler actively processing.

Note, while handlers work concurrently, there's only ever a single handler in flight for any given stream; it gets passed the queue of all pending events for that stream 

(cant think of a good sentence to convey "and you can do them sequentially, run them in parallel, do batch SQL", without contention knowing that there are not other events for that stream being handled at the same time)